### DISCUSSION TITLE
how do i style with css ?

### DISCUSSION THREAD

**Discussion by kwatman** [Mogztter/asciidoctor-web-pdf#441] [Q&A]
hello evryone im new to asciidocs pdf and i want to use it for a project. so far everything works fine. but nowi want to change how evrything looks. i know this can be done with css. but what classes can i acces to overide ?

**Comment by ggrossetie** [comment]
Hello @dolfijn3000 

Currently, we rely on the built-in HTML 5 converter from Asciidoctor.
So the easiest way is probably to take a look at the default stylesheet from Asciidoctor: https://github.com/asciidoctor/asciidoctor/blob/master/data/stylesheets/asciidoctor-default.css

> but what classes can i acces to overide ?

It really depends on what you want to override. Each AsciiDoc blocks and elements produce a specific HTML structure that can have default roles (i.e., CSS classes).

Here's an example:

```
.Optional title
[example.myrole]
This is an example of an example block.
```

For reference, the above AsciiDoc document will produce the following HTML:

```html
<div class="exampleblock myrole">
<div class="title">Example 1. Optional title</div>
<div class="content">
This is an example of an example block.
</div>
</div>
```

I would recommend using the `--preview` option to inspect the HTML produced using Chrome DevTools.

Please note that you can also extend the default converter to change how blocks and elements are converted to HTML: https://github.com/Mogztter/asciidoctor-web-pdf#custom-layout

I hope it helps!

**Comment by wh81752** [comment]
@dolfijn3000 - Invoke asciidoctor-web-pdf to generate PDF and HTML. Then open generated HTML in your ChromeBrowser to inspect the document (use https://developer.chrome.com/docs/devtools/). 

By "pointing" at an HTML you can see which CSS rules apply. You can also change CSS rules to test what kind of effect it has. DevTools also tells you where the CSS rules are coming from (buildin, from your CSS file, imported,  whatever).

You can use option --watch to auto-generate PDF / HTML if your input document changes (==> not sure whether asciidoc-web-pdf also watches your CSS files, I guess it does).

You may also want to use --preview which automatically opens your HTML document in Chromium (stripped down version of ChromeBrowser suited for (automated) web development).

**Comment by wh81752** [comment]
> Then open generated HTML in your ChromeBrowser to inspect the document (use https://developer.chrome.com/docs/devtools/).

Open generated HTML in your local browser works fine as long you are not loading something from your local computer (for example customized CSS). When doing so your local script gets blocked nowadays due to CORS. To get away you 

1. use option --preview ; or
2. use a local http-server to render your HTML in your browser
